Input/InputFieldSpotlightDecorator.js

```diff
@@ -141,78 +141,84 @@ const InputSpotlightDecorator = hoc(defaultConfig, (config, Wrapped) => {
 		constructor (props) {
 			super(props);
 
-			this.state = {
-				focused: null,
-				node: null,
-				fromMouse: false
-			};
-
+			this.focused = null;
+			this.node = null;
+			this.fromMouse = false;
+			this.prevFocused = null;
+			this.prevNode = null;
 			this.ariaHidden = props.noReadoutOnFocus || null;
 			this.paused = new Pause('InputSpotlightDecorator');
 			this.handleKeyDown = handleKeyDown.bind(this);
 		}
 
-		componentDidUpdate (_, prevState) {
-			this.ariaHidden = null;
-			this.updateFocus(prevState);
-		}
-
 		componentWillUnmount () {
 			this.paused.resume();
 
-			if (this.state.focused === 'input') {
+			if (this.focused === 'input') {
 				const {onSpotlightDisappear} = this.props;
 
 				if (onSpotlightDisappear) {
 					onSpotlightDisappear();
 				}
 
 				if (!noLockPointer) {
-					releasePointer(this.state.node);
+					releasePointer(this.node);
 				}
 			}
 		}
 
-		updateFocus = (prevState) => {
+		updateComponent () {
```

Comment:
Can't we keep the previous interface?
updateFocus = (prevState)?

Is ther any reason that we should hav two function `updateComponent` and `updateFocus`?

Reply:
I fixed it.